test/integration/cni/pod_traffic_across_az_test.go

```diff
@@ -0,0 +1,305 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		By("deleting the Daemonset.")
+		err = f.K8sResourceManagers.DaemonSetManager().DeleteAndWaitTillDaemonSetDeleted(testDaemonSet, utils.DefaultDeploymentReadyTimeout)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("While testing connectivity across AZ", func() {
+
+		BeforeEach(func() {
+			serverPort = 2273
+			protocol = ec2.ProtocolTcp
+			// Test tcp connection using netcat
+			serverListenCmd = []string{"nc"}
+			// The nc flag "-l" for listen mode, "-k" to keep server up and not close
+			// connection after each connection
+			serverListenCmdArgs = []string{"-k", "-l", strconv.Itoa(serverPort)}
+			// netcat verbose output is being redirected to stderr instead of stdout
+			testerExpectedStdErr = "succeeded!"
+			testerExpectedStdOut = ""
+
+			// The nc flag "-v" for verbose output and "-wn" for timing out in n seconds
+			testConnectionCommandFunc = func(receiverPod coreV1.Pod, port int) []string {
+				return []string{"nc", "-v", "-w2", receiverPod.Status.PodIP, strconv.Itoa(port)}
+			}
+
+			// Create a negative test case with the wrong port number. This is to reinforce the
+			// positive test case work by verifying negative cases do throw error
+			testFailedConnectionCommandFunc = func(receiverPod coreV1.Pod, port int) []string {
+				return []string{"nc", "-u", "-v", "-w2", receiverPod.Status.PodIP, strconv.Itoa(port + 1)}
+			}
+		})
+
+		It("Should allow TCP traffic across AZs.", func() {
+			CheckConnectivityBetweenPods(azToPod, serverPort, testerExpectedStdOut, testerExpectedStdErr, testConnectionCommandFunc)
+			VerifyConnectivityForNegativeCase(azToPod, serverPort, testFailedConnectionCommandFunc)
+		})
+	})
+})
+
+var _ = Describe("[STATIC_CANARY] API Server Connectivity from AZs", FlakeAttempts(3), func() {
+
+	var (
+		err           error
+		testDaemonSet *v1.DaemonSet
+
+		// Map of Pods placed on primary/secondary ENI IP
+		interfaceToPodList common.InterfaceTypeToPodList
+
+		azName string
+
+		// Map of AZ name to Pod
+		azToPod map[string]coreV1.Pod
+	)
+
+	JustBeforeEach(func() {
+		serverContainer := manifest.
+			NewCurlContainer().
+			Command([]string{
+				"sleep",
+				"3600",
+			}).
+			Build()
+
+		By("creating a server DaemonSet on primary node")
+
+		testDaemonSet = manifest.
+			NewDefaultDaemonsetBuilder().
+			Container(serverContainer).
+			PodLabel("role", "az-test").
+			Name("api-server-connectivity-daemonset").
+			Build()
+
+		_, err = f.K8sResourceManagers.DaemonSetManager().CreateAndWaitTillDaemonSetIsReady(testDaemonSet, utils.DefaultDeploymentReadyTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		By(fmt.Sprintf("getting the node with the node label key %s and value %s",
+			f.Options.NgNameLabelKey, f.Options.NgNameLabelVal))
+
+		nodes, err := f.K8sResourceManagers.NodeManager().GetNodes(f.Options.NgNameLabelKey, f.Options.NgNameLabelVal)
+		Expect(err).ToNot(HaveOccurred())
+
+		azToPod = make(map[string]coreV1.Pod)
```

Comment:
If we build a `nodeToAz` or `azToNode` map, then we can deploy pods using node selectors in a more targeted fashion

Reply:
Replied above. I was thinking of node selector. But daemonset seems simpler.

When the cluster create setup (separate from this script), creates nodes across AZs, we don't need to have those logic with node selector.

Reply:
Yeah, I agree. Daemonset is probably the simplest way to ensure our test pods are launched on all the nodes. Do we ensure we only have 1 worker node per AZ in our setup? @orsenthil